boeing_voucher/pdf_parsing.py

```diff
@@ -0,0 +1,153 @@
+                if column_values[column]:
+                    column_values[column] += " "
+                column_values[column] += value
+
+
+def read_pdf_international(pdf):
+    """
+    Process international Boeing vouchers and return parsed values
+    """
+    row_values = {
+        "Employee Name": "",
+        "Program": "",
+        "Course of Study": "",
+        "School": "",
+        "BEMS ID": "",
+        "Pre-Authorized Amount": "",
+        "Dates": "",
+        "Class or Module Title": "",
+        "Class or Module Number": "",
+    }
+    extend_values = {}
+
+    previous_line = ""
+    recording_address = False
+    for page in pdf:
+        for line in page.splitlines():
+            if line.startswith("To Whom It"):
+                extend_values["Date"] = previous_line.strip()
+            if line.startswith("Entity Name:"):
+                extend_values["Entity Name"] = previous_line.strip()
+
+            if line.startswith("School Instructions:"):
+                recording_address = False
+            if recording_address and "______" not in line:
+                extend_values["Address"] += " " + line.strip()
+            if line.startswith("Address:"):
+                recording_address = True
+                extend_values["Address"] = line.replace("Address:", "").strip()
+
+            for row_name in row_values:
+                if line.startswith(row_name):
+                    row_values[row_name] = line.replace(row_name + ":", "").strip()
+            previous_line = line
+    row_values.update(extend_values)
+    return row_values
+
+
+def read_pdf(pdf_file):
+    """
+    Take in Boeing Voucher PDFs and parse them as either international or domestic, and return unified response.
+    """
+    try:
+        pdf = pdftotext.PDF(pdf_file)
+        if any("Entity Name:" in page for page in pdf):
+            values = read_pdf_international(pdf)
+            return {
+                "BEMSID": values.get("BEMS ID"),
+                "voucher_id": None,
+                "course_start_date": values.get("Dates").split(" ")[0],
+                "class_module_number": values.get("Class or Module Number"),
+                "class_module_title": values.get("Class or Module Title"),
+                "employee_name": values.get("Employee Name"),
+            }
+        else:
+            values = read_pdf_domestic(pdf)
+            return {
+                "BEMSID": values.get("BEMSID"),
+                "voucher_id": values.get("Voucher ID"),
+                "course_start_date": values.get("Course Dates").split(" ")[0],
+                "class_module_number": values.get("Course Name").split(" ")[0],
+                "class_module_title": " ".join(
+                    values.get("Course Name").split(" ")[1:]
+                ),
+                "employee_name": values.get("Employee Name"),
+            }
+    except Exception:  # pylint: disable=broad-except
+        log.error("Could not parse PDF")
```

Comment:
Can you add a unit test that will trigger this exception?

Reply:
Added a test for this.